flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/sink/AbstractFlinkSink.java

```diff
@@ -158,4 +125,10 @@ private void assertCheckpointConfiguration(StreamExecutionEnvironment env) {
                         + ExecutionCheckpointingOptions.CHECKPOINTING_MODE.key()
                         + " to exactly-once");
     }
+
+    protected abstract OneInputStreamOperator<RowData, Committable> createWriteOperator(
+            String initialCommitUser, boolean isStreaming);
+
+    protected abstract OneInputStreamOperator<Committable, Committable> createCommitterOperator(
```

Comment:
Here we can just a `CommitRetry`, let subclasses only return different ones and mask the same part.